pkg/policy/api/l4.go

```diff
@@ -150,6 +150,19 @@ type Listener struct {
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:Required
 	Name string `json:"name"`
+
+	// Priority is used to select the Listener if multiple rules would apply different listeners
+	// to a policy map entry. Lower numbers indicate higher priority. If left out, a random, but
+	// unique priority on the range 10000-20000 is assigned, so that traffic will be
+	// consistently redirected to the same listener. If higher priority desired, a low unique
+	// number like 1, 2, or 3 should be explicitly specified here. If a lower than default
+	// priority is needed, then a unique number higher than 20000 should be explicitly
+	// specified. Numbers on the default range (10000-20000) are not allowed.
+	//
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:validation:Optional
+	Priority uint16 `json:"priority"`
```

Comment:
Let's say the cluster administrator creates a listener policy in a CCNP for a particular port, then a namespace user creates another listener policy in CNP for a port, I guess that means that the namespace user can override/preempt the clusterwide configuration? I'm trying to think a little bit outside the box about how the semantics of this field may apply in conjunction with other policy primitives we have.

One might argue that if the cluster administrator wants to ensure that a particular listener applies on a port, then they could do so with priority 1, and if they wanted to provide a way for it to be overridden, they could select a more deprioritized value. But then, what happens if both policies have priority 1? I guess we fall back to random / implementation-defined behaviour?

Reply:
Interesing aspect that warrants further thought. Currently, any deny policies in a CCNP get precedence over any allows in a CNP (like they do in a single CNP). Maybe listener references in CCNPs should get precedence over those in CNPs? This also points to the direction that we can't nail down the API spec for this at this moment, but must leave it open for improvement.

Reply:
Do we just implement this by parsing the priorities and adding an offset to them for CNP?

Ie if there's a CCNP with a priority, it gets the priority specified by the user, maybe bounded 1-100 for example. If there's a CNP, after Cilium parses the priority it adds the max value (100) so the CNP gets priority 101 for user input `priority: 1` etc.

Or for now we just leave it undefined until we have a stronger set of user stories to back up the API feature.

Reply:
Would that impose a limit of max 99 CNPs with listener priority?

Reply:
For that proposal it would not limit the number of CNPs that someone can create. We would still need to define that if there are two CNPs selecting the same traffic with the same listener priority then the behaviour is undefined. All it would do is impose an ordering between CCNP vs. CNP.